net.go

```diff
@@ -109,24 +110,17 @@ func (d *Dialer) Dial(remote ma.Multiaddr) (Conn, error) {
 	// ok, Dial!
 	var nconn net.Conn
 	switch rnet {
-	case "tcp", "tcp4", "tcp6":
+	case "tcp", "tcp4", "tcp6", "udp", "udp4", "udp6":
 		nconn, err = d.Dialer.Dial(rnet, rnaddr)
 		if err != nil {
 			return nil, err
 		}
-	case "udp", "udp4", "udp6":
-		return nil, fmt.Errorf("utp is currently broken")
-
-		// // construct utp dialer, with options on our net.Dialer
-		// utpd := utp.Dialer{
-		// 	Timeout:   d.Dialer.Timeout,
-		// 	LocalAddr: d.Dialer.LocalAddr,
-		// }
-		//
-		// nconn, err = utpd.Dial(rnet, rnaddr)
-		// if err != nil {
-		// 	return nil, err
-		// }
+	case "utp", "utp4", "utp6":
+		// construct utp dialer, with options on our net.Dialer
+		nconn, err = utp.DialTimeout(rnaddr, d.Timeout)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I think this wants a `Dialer` too, like the other `utp` package. the interface is on the left, commented out:

```go
utpd := utp.Dialer{
  Timeout: d.Dialer.Timeout,
  LocalAddr: d.Dialer.LocalAddr,
}

nconn, err = utpd.Dial(rnet, rnaddr)
if err != nil {
  return nil, err
}
```
it makes it much easier to map other options (like SetDeadline, and so on) if all packages follow a similar set of interfaces as the `net` package. i know anacrolix/utp doesn't, but we can map.